src/scmdata/netcdf.py

```diff
@@ -249,20 +257,12 @@ def _read_var(name, var):
                         var.dimensions, meta_at_coord[(slice(None),) + it.multi_index]
                     )
                 }
-                if is_scmdf:
-                    for c, v in meta_vals.items():
-                        df.loc[df[c] == v, col] = values[it.multi_index]
-
-                else:
-                    df.filter(**meta_vals)[col] = values[it.multi_index]
-
-        if is_scmdf:
-            df.loc[:, col] = df.loc[:, col].astype(values[0].dtype)
-
-    if is_scmdf:
-        df = ScmDataFrame(df)
-    else:
-        df.metadata.update({k: ds.getncattr(k) for k in ds.ncattrs()})
+                df[col] = _merge_meta(df.meta, col, values[it.multi_index], **meta_vals)
+        try:
+            df[col] = df[col].astype(var.dtype)
+        except ValueError:
+            logger.exception("could not convert meta dtype")
```

Comment:
will the reading still work or will it raise somewhere?

Reply:
It was an emergency valve. I might have to add some additional tests as I'm not sure if there is a way to make it fail.